### DISCUSSION TITLE
sleep(2) in multi_gpu_test function.

### DISCUSSION THREAD

**Discussion by eunwoosh** [open-mmlab/mmclassification#1175] [General] [answered]
https://github.com/open-mmlab/mmclassification/blob/9b1cd5c3b1c9a307361aa7a71c2d1b611053c7dd/mmcls/apis/test.py#L116

Greetings,

I wonder that sleep(2) should be necessary in multi_gpu_test function.
It seems that process sleeps 2 seconds every validation step, which make training quite slow.
Is there any possibility to meet deadlock if this line is removed?

Thank you.

**Comment by Ezra-Yu** [comment]
refer to https://github.com/open-mmlab/mmclassification/pull/666

**Comment by mzr1996** [comment, marked as answer]
Small magic, this issue is discussed in detailed at https://github.com/open-mmlab/mmcv/issues/1640.
I think you can also remove this line if you don't meet the deadlock problem.

**Comment by eunwoosh** [comment]
To summarize, deadlock can occur during dataloader switch and the reason is too many threads compared to CPU cores.
Thank you who comments! it can be better to deal with this problem not using sleep() as mentioned in other discussion.
